azkaban-common/src/main/java/azkaban/user/FileWatcher.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.user;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.List;
+
+public class FileWatcher {
+
+  private final WatchService watchService;
+  private final SensitivityWatchEventModifier sensitivity;
+
+  public FileWatcher() throws IOException {
```

Comment:
The idea is that you can "just use it", expecting that the default sensitivity is good for the typical use case. This also allows using the simple method reference `::new` for the factory lambda. Otherwise a static method should be created for that purpose, but constructor is more clearly scoped.